crypto/bls/bls.go

```diff
@@ -158,3 +164,32 @@ func VerifySignature(publicKey []byte, message []byte, extraData []byte, signatu
 	err = publicKeyObj.VerifySignature(message, extraData, signatureObj, shouldUseCompositeHasher)
 	return err
 }
+
+func EncodeEpochSnarkData(newValSet []SerializedPublicKey, maximumNonSignersPlusOne uint32, epochIndex uint16) ([]byte, error) {
```

Comment:
What is `maximumNonSignersPlusOne` used for? Note that in some cases, `F+1 != ValSet.Size() - MinQuorumSize() + 1`, e.g. a val set of size 9 will have a min quorum size of 6 but an F of 2.

Reply:
The SNARK needs it, but your way remind me it was the wrong way to do it. Thank.